common-money.lic

```diff
@@ -40,26 +40,45 @@ module DRCM
     check_wealth(hometown_currency(hometown))
   end
 
-  def ensure_copper_on_hand(copper, hometown)
-    on_hand = wealth(hometown)
+  def ensure_copper_on_hand(copper, settings)
+    on_hand = wealth(settings.hometown)
     return true if on_hand >= copper
     withdrawals = minimize_coins(copper - on_hand)
-    withdrawals.all? { |amount| withdraw_exact_amount?(amount, hometown) }
+    withdrawals.all? { |amount| withdraw_exact_amount?(amount, settings) }
   end
 
-  def withdraw_exact_amount?(amount_as_string, hometown)
-    bank_id = get_data('town')[hometown]['deposit']['id']
+  def withdraw_exact_amount?(amount_as_string, settings)
+    if settings.bankbot_enabled
+      bank_id = settings.bankbot_room_id
+    else  
+      bank_id = get_data('town')[settings.hometown]['deposit']['id']
+    end  
     DRCT.walk_to(bank_id)
     DRC.release_invisibility
     loop do
-      case DRC.bput("withdraw #{amount_as_string}", 'The clerk counts', 'The clerk tells', 'The clerk glares at you.', 'You count out', 'find a new deposit jar', 'If you value your hands', 'Hey!  Slow down!', "You must be at a bank teller's window to withdraw money")
-      when 'The clerk counts', 'You count out'
-        break true
-      when 'The clerk glares at you.', 'Hey!  Slow down!'
-        pause 15
-      when 'The clerk tells', 'If you value your hands', 'find a new deposit jar', "You must be at a bank teller's window to withdraw money"
-        break false
-      end
+      if settings.bankbot_enabled && DRRoom.pcs.include?(settings.bankbot_name)
+        amount_convert, type = amount_as_string.split(" ")
```

Comment:
Split on " " is the same as calling split
```
irb(main):001:0> "hello world".split
=> ["hello", "world"]
irb(main):001:0> "hello world".split(" ")
=> ["hello", "world"]
```

Reply:
I can make less explicit. :)